### PR TITLE
chore: clean up and update dependencies

### PR DESCRIPTION
**Pull request by solarw**
clean up dependencies
set latest versions

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -57,40 +57,33 @@
...
         "bech32",
         "requests",
-        "google-api-python-client",
         "protobuf>=3.19.4,<4",
-        "python-mbedtls==2.2.0",
-        "grpcio",
-        "click",
-        "types-certifi",
+        "grpcio==1.47.0",
         "bip-utils",
```
Comment (5A11) [inline setup.py:62]: Do we need `bip-utils`?

File: setup.py [hunk 2]
```diff
@@ -57,40 +57,33 @@
...
         "blspy",
+        "google-api-python-client",
```
Comment (5A11) [inline setup.py:64]: Do we need `google-api-python-client`?

File: cosmpy/crypto/hashfuncs.py [hunk 3]
```diff
@@ -21,15 +21,8 @@
...
-
-# Detect if ripemd160 can actually be used in the system. Querying `hashlib.algorithms_available`
-# does not mean much and will fail on 22.04 LTS
-try:
-    hashlib.new("ripemd160")
-    _ripemd160_present = True
-except ValueError:
-    _ripemd160_present = False
+# pycryptodome, a dependency of bip-utils
+from Crypto.Hash import RIPEMD160  # type: ignore # nosec
```
Comment (5A11) [inline cosmpy/crypto/hashfuncs.py:25]: Need to test this on an M1/M2 Mac. I think there was a problem on an M1 Mac which lead to the existing solution.

File: setup.py [hunk 4]
```diff
@@ -57,40 +57,33 @@
...
-            "flake8-copyright==0.2.2",
-            "flake8-bugbear==22.7.1",
-            "flake8-eradicate==1.2.1",
-            "flake8-docstrings==1.6.0",
-            "pydocstyle==6.1.1",
+            "vulture==2.5",
+            "pylint==2.14.5",
+            "liccheck==0.7.2",
+            "flake8-copyright==0.2.3",
+            "grpcio==1.47.0",
```
Comment (5A11) [inline setup.py:82]: This is in the standard dependencies.

File: setup.py [hunk 5]
```diff
@@ -57,40 +57,33 @@
...
+            "safety==2.1.1",
+            "isort==5.10.1",
             "darglint==1.8.1",
-            "vulture==2.3",
-            "pylint==2.9.6",
-            "liccheck==0.6.2",
-            "flake8-copyright==0.2.2",
-            "flake8-bugbear==22.7.1",
-            "flake8-eradicate==1.2.1",
-            "flake8-docstrings==1.6.0",
```
Comment (5A11) [inline setup.py:88]: Why remove these three?

File: setup.py [hunk 6]
```diff
@@ -57,40 +57,33 @@
...
-            "pydocstyle==6.1.1",
```
Comment (5A11) [inline setup.py:89]: Why remove these 4?

File: strategy.ini [hunk 7]
```diff
@@ -87,3 +87,5 @@ unauthorized_licenses:
 [Authorized Packages]
 pathspec: >=0.9.0
 certifi: >=2019.11.28
+blspy: *
+pycryptodome: *
```
Comment (5A11) [inline strategy.ini:91]: What is this list?

File: setup.py (hunk 2 above)
Comment (solarw) [reply setup.py:64]: yes, it's required by grpc

File: setup.py [hunk 8]
```diff
@@ -57,40 +57,33 @@
...
+            "vulture==2.5",
+            "pylint==2.14.5",
+            "liccheck==0.7.2",
+            "flake8-copyright==0.2.3",
+            "grpcio==1.47.0",
+            "importlib-metadata==4.3.0",
```
[the remark this answers is not in the document]
Comment (solarw) [reply setup.py]: yes, was for testing

File: setup.py (hunk 4 above)
Comment (solarw) [reply setup.py:84]: fixed

File: setup.py (hunk 6 above)
Comment (solarw) [reply setup.py:89]: added back

File: strategy.ini (hunk 7 above)
Comment (solarw) [reply strategy.ini:91]: liccheck fails on those two packages

File: setup.py [hunk 9]
```diff
@@ -57,40 +57,37 @@
...
             "check-manifest",
-            "tox==3.24.1",
+            "tox==3.25.1",
             "flake8==3.9.2",
-            "black==22.3",
-            "mypy==0.910",
+            "black==22.6",
+            "mypy==0.971",
             "mkdocs==1.3",
+            "markdown==3.3.4",
```
Comment (5A11) [inline setup.py:74]: Do we need "markdown" specifically added? I think it is a dub-dependency of material-for-mkdocs.

File: setup.py [hunk 10]
```diff
@@ -57,40 +57,37 @@
...
             "mkdocs-material==8.2.11",
```
Comment (5A11) [inline setup.py:75]: Can you bump to `8.4.0`?

File: setup.py (hunk 9 above)
Comment (solarw) [reply setup.py]: done

File: setup.py (hunk 10 above)
Comment (solarw) [reply setup.py]: done

**Review by 5A11 — APPROVED** [review]
lgtm

**Pull request merged**